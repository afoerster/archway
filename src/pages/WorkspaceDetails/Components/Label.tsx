import * as React from 'react';

interface Props {
  style?: React.CSSProperties;
  children: any;
}

const Label = ({ style, children }: Props) => (
  <div
    style={{
        marginBottom: 10,
        fontSize: 16,
        textTransform: 'uppercase',
        letterSpacing: 1,
        fontWeight: 200,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        ...style,
      }}>
    {children}
  </div>
);

export default Label;