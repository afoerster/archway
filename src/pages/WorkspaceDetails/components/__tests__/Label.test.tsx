import * as React from 'react';
import { shallow } from 'enzyme';

import Label from '../Label';

describe('Label', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Label children={null} />);
    expect(wrapper).toMatchSnapshot();
  });
});
