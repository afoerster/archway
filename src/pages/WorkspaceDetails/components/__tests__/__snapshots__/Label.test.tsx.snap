// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": null,
      "style": Object {
        "alignItems": "center",
        "display": "flex",
        "fontSize": 16,
        "fontWeight": 200,
        "justifyContent": "center",
        "letterSpacing": 1,
        "marginBottom": 10,
        "textTransform": "uppercase",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "style": Object {
          "alignItems": "center",
          "display": "flex",
          "fontSize": 16,
          "fontWeight": 200,
          "justifyContent": "center",
          "letterSpacing": 1,
          "marginBottom": 10,
          "textTransform": "uppercase",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
